Client/Frontend/Browser/Playlist/Controllers/PlaylistCarplayController.swift

```diff
@@ -63,38 +62,49 @@ class PlaylistCarplayController: NSObject {
     }
     
     func observeFolderStates() {
-        folderObserver = PlaylistManager.shared.onCurrentFolderDidChange
-            .receive(on: DispatchQueue.main)
-            .sink { [weak self] in
-                guard let self = self else { return }
-                
-                if self.interfaceController.rootTemplate == self.interfaceController.topTemplate {
-                    // Nothing to Pop
+        PlaylistCarplayManager.shared.onCarplayUIChangedToRoot.eraseToAnyPublisher()
+        .receive(on: DispatchQueue.main)
+        .sink { [weak self] in
+            guard let self = self else { return }
+            self.interfaceController.popToRootTemplate(animated: true) { success, error in
+                if !success, let error = error {
+                    log.error(error)
+                }
+            }
+        }.store(in: &playlistObservers)
+        
+        PlaylistManager.shared.onCurrentFolderDidChange
+        .receive(on: DispatchQueue.main)
+        .sink { [weak self] in
+            guard let self = self else { return }
+            
+            if self.interfaceController.rootTemplate == self.interfaceController.topTemplate {
+                // Nothing to Pop
+                // Push the Folder Template
+                let listTemplate = self.generatePlaylistListTemplate()
+                self.interfaceController.pushTemplate(listTemplate, animated: true) { [weak self] success, error in
+                    if !success, let error = error {
+                        self?.displayErrorAlert(error: error)
+                    }
+                }
+                return
+            }
+            
+            // Pop to Root Template
+            self.interfaceController.popToRootTemplate(animated: false) { success, error in
+                if !success, let error = error {
```

Comment:
Is there possibility situation where action is not successful but error is empty?

Reply:
Not possible. But also wouldn't matter because we're only logging it anyway. We don't display this to the user as it won't have any useful information and it's a system error.

Reply:
I asked it because if it not a success and error is nil, this will try to push the folder template but since you said it is not possible we are good here.